### PR TITLE
Create *TreeImpl for accessor and generator method

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java [hunk 1]
```diff
@@ -74,6 +76,21 @@ public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
     scan(tree.body().statements());
   }
 
+  @Override
+  public void visitAccessorMethodDeclaration(AccessorMethodDeclarationTree tree) {
+    scan(tree.name());
+    scan(tree.parameters());
+    // Ignoring empty function
```
Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java]: @vilchik-elena  why so ?

File: javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    this.body = body;
+  }
+
+  public GeneratorMethodDeclarationTreeImpl completeWithStaticToken(InternalSyntaxToken staticToken) {
+    this.staticToken = staticToken;
+
+    return this;
+  }
+
+  @Nullable
```
Comment (lindamartin) [inline javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java]: @vilchik-elena I guess this one could be removed as starToken should not be nullable at this point, should it ?

File: javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java [hunk 3]
```diff
@@ -45,4 +46,5 @@
 
   BlockTree body();
 
+  MethodDeclarationTree completeWithStaticToken(InternalSyntaxToken staticToken);
```
Comment (lindamartin) [inline javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java]: @vilchik-elena I don't think that this should appear in the interface it is a implementation detail and should not be visible on an API level.

**Comment by lindamartin** [comment]
See my comments, otherwise the rest looks much better now!

File: javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java (hunk 1 above)
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java]: @lindamartin I just commented code of visitMethodDeclaration (as previously accessors and generators were processed in this method). As i understand, this rule shouldn't raise issues for empty function bodies, so in next line we `scan(tree.body().statements());` and not `scan(tree.body());`.

File: javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java (hunk 2 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java]: @lindamartin this `@Nullable` belongs to staticToken

File: javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java (hunk 1 above)
Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/EmptyBlockCheck.java]: @vilchik-elena indeed, got it! I thought it was in BaseTreeVisitor, obviously my eyes crossed ^^

File: javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java (hunk 3 above)
Comment (vilchik-elena) [inline javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java]: @lindamartin indeed. I removed this method completely :) 51dc6eaedc79c40ad653dde52bbe81b24045053a

File: javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java (hunk 2 above)
Comment (lindamartin) [inline javascript-frontend/src/main/java/org/sonar/javascript/tree/impl/declaration/GeneratorMethodDeclarationTreeImpl.java]: @vilchik-elena indeed my bad I misread again

File: javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java (hunk 3 above)
Comment (lindamartin) [inline javascript-frontend/src/main/java/org/sonar/plugins/javascript/api/tree/declaration/MethodDeclarationTree.java]: @vilchik-elena cool, even if it seems redundant, it looks clearer in the grammar IMHO :)

**Pull request merged**